### DISCUSSION TITLE
Allow editing on the system fields

### DISCUSSION THREAD

**Discussion by Attacler** [directus/directus#19310] [Feature Requests]
### Summary

For most of my projects i wish that i can reorder/hide or use conditions on the fields of the system collections.
This is because some of the fields are not relevant or could be positioned better between custom fields.

### Basic Example

This does not include any new or changed API`s for as far as i know.

### Motivation

For some projects not all fields are needed, it would be great if we can chage the visual aspects of those fields.
For example: avatar/location/title/description

### Detailed Design

This would have the enable to handle any frontend/ui actions on the fields.
Except the editing or removing of a fields, because this can break the app.

### Must Have:
- Edit support for the field/interface/display/validations/conditions
- Edit support for "Hide on Detail"
- Allow reordering of fields in the system collections

### Should Have:
- 

### Could Have:
- Edit support for non visible system collections (for example directus_sessions)

### Won't Have:
- Delete(/edit support) for DB level configurations since this would break the data studio

### Drawbacks

I dont think any apply because the functionallity is already there.
Only draw back is that any updates that changes the visual part will effect users (resetting their settings on that field).

There might be more drawbacks, right now it seems like the migrations contain a locked: true which decides what fields can be edited and which cannot.

### Alternatives

Right now the only way to kinda achive this is by using css rules

### Adoption Strategy

I dont think its breaking for the users, only for the Directus team and maybe future updates,

### Unresolved Questions

Its my first feature request in this format, please let me know if i need to add more :)

**Comment by rijkvanzanten** [comment]
> Its my first feature request in this format, please let me know if i need to add more :)

This is perfect 🙌🏻

**Comment by LudovicSterlin** [comment]
Hello, any new on this subject ? Or is there a place where we can find what is the status of this on Directus side ? 
It has 36 upvotes 🙏 
Thanks for your work !

**Comment by liferacer23** [comment]
Has anyone solved this

**Comment by cicunam** [comment]
one more vote

**Reply by smile2008** [reply]
Just another vote

**Comment by rafarc21** [comment]
Did anyone found a workaround for this? perhaps custom css? 

I'm going to experiment with this extension to see if it works: 
https://github.com/brunribeiro/directus-user-field-hider

**Reply by rijkvanzanten** [reply]
Custom css — while hacky — is the way to go here. The system fields are not in the database — they're statically in the code — which means they're not editable like regular fields at this point 👍🏻

**Comment by IamSAB** [comment]
Here you find a guide to add them to the database to make them configurable:
https://www.datac.com/posts/customizing-directus-default-fields

However, I do not understand, why these fields are not configurable in the first place?

Couldmt we just disable the whole view and only let them edit if the user gives consent that editing the fields can break the system. Or as proposed before, just not let them get deleted.

But in any case, hide or rearrange them would be essential for everyone I think!

**Reply by rijkvanzanten** [reply]
> However, I do not understand, why these fields are not configurable in the first place?

Because the system field metadata isn't managed in the database, but is done statically through yaml files in here: https://github.com/directus/directus/blob/main/packages/system-data/src/fields/users.yaml

This is because we often have to make changes to these system metadata files, which causes a lot of additional migrations and chances of conflict when we're trying to change something that the user has already customized as well 👍🏻

**Comment by domdus** [comment]
Check out "System Fields Manager" in the Marketplace ✌️

Or the repo:
https://github.com/domdus/directus-extension-system-fields-manager

**Reply by LudovicSterlin** [reply]
Looks nice, I will not be able to test my need anymore but that would have been nice at that time
